pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -80,6 +80,49 @@ func NewNodeController(
 	}
 }
 
+// Generates num pod CIDRs that could be assigned to nodes.
+func generateCIDRs(num int) util.StringSet {
+	res := util.NewStringSet()
+	for i := 0; i < num; i++ {
+		// TODO: Make the CIDRs configurable.
+		res.Insert(fmt.Sprintf("10.244.%v.0/24", i))
+	}
+	return res
+}
+
+// For each node from newNodes, finds its current spec in registeredNodes.
+// If it is not there, it gets a new valid CIDR assigned.
+func reconcilePodCIDRs(newNodes, registeredNodes *api.NodeList) *api.NodeList {
```

Comment:
The pattern is often used when you want to one-line a call.

```
var foo Foo
DoSomething(&foo)
OtherThing(&foo)
```

Vs

```
var foo Foo
OtherThing(DoSomething(&foo))
```
On Mar 24, 2015 6:42 PM, "CJ Cullen" <notifications@github.com> wrote:

> In pkg/cloudprovider/controller/nodecontroller.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5246#discussion_r27089504>
> :
>
> > @@ -80,6 +80,49 @@ func NewNodeController(
> >  	}
> >  }
> >
> > +// Generates num pod CIDRs that could be assigned to nodes.
> > +func generateCIDRs(num int) util.StringSet {
> > +	res := util.NewStringSet()
> > +	for i := 0; i < num; i++ {
> > +		// TODO: Make the CIDRs configurable.
> > +		res.Insert(fmt.Sprintf("10.244.%v.0/24", i))
> > +	}
> > +	return res
> > +}
> > +
> > +// For each node from newNodes, finds its current spec in registeredNodes.
> > +// If it is not there, it gets a new valid CIDR assigned.
> > +func reconcilePodCIDRs(newNodes, registeredNodes *api.NodeList) *api.NodeList {
>
> Yes. I'm saying you don't need to:
> func bar(
>
> *foo) *foo where the pointer doesn't actually change can just be: func
> bar(*foo)
>
> I've seen this pattern elsewhere too. Is there some good reason for it
> that I am missing?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5246/files#r27089504>
> .
>